Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.MapViewModel.swift

```diff
@@ -55,10 +67,93 @@ extension OfflineMapAreasView {
             }
         }
         
-        /// Requests authorization to show notifications.
-        nonisolated func requestUserNotificationAuthorization() async {
-            _ = try? await UNUserNotificationCenter.current()
-                .requestAuthorization(options: [.alert, .sound])
+        /// Makes offline preplanned map models with information from the downloaded mobile map
+        /// packages for the online map.
+        func makeOfflinePreplannedMapModels() async {
+            guard let portalItemID else { return }
+            
+            let preplannedDirectory = FileManager.default.preplannedDirectory(forPortalItemID: portalItemID)
+            
+            guard let mapAreaIDs = try? FileManager.default.contentsOfDirectory(atPath: preplannedDirectory.path()) else { return }
```

Comment:
[`contentsOfDirectory(at URL)`](https://developer.apple.com/documentation/foundation/filemanager/1413768-contentsofdirectory) equivalent of this method can be used.

Reply:
I did not apply this suggestion since it results in `mapAreaIDs` being an array of `URL` instead of `String`. `String` is more convenient to work with in this case.